pytorch_lightning/trainer/connectors/model_connector.py

```diff
@@ -45,7 +45,10 @@ def copy_trainer_model_properties(self, model):
             m.use_ddp2 = self.trainer.use_ddp2
             m.use_ddp = self.trainer.use_ddp
             m.use_amp = self.trainer.amp_backend is not None
-            m.testing = self.trainer.testing
+            # TODO: I only find usages of m.testing in DDP, where it's used to
+            #  discriminate test from validation, as opposed to test from fit in
+            #  Trainer. Still need to fully determine if it's correct.
+            m.testing = self.trainer.evaluating == 'test'
```

Comment:
It seems to me that `m.testing` is interpreted a bit differently than `trainer.testing`:
* The latter is used in most of the code to discriminate if the top-level function that the user called was `fit()` or `test()`. As such, in the rest of the code I replaced `if trainer.testing` with `if trainer.evaluating`, that is `validate()` and `test()` take the same code path
* The former is used by DP and DDP to discriminate if the model is currently inside the validation loop or the test loop (eg. [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/0209cfc2f20534330546d054870d8ddbc824a483/pytorch_lightning/overrides/data_parallel.py#L262))

If this interpretation is correct, the code should be good as is. If not, it needs to be changed.